x/tokenfactory/keeper/msg_server.go

```diff
@@ -0,0 +1,97 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	grpccodes "google.golang.org/grpc/codes"
+	grpcstatus "google.golang.org/grpc/status"
+
+	"github.com/NibiruChain/nibiru/x/tokenfactory/types"
+)
+
+var _ types.MsgServer = (*Keeper)(nil)
+
+var errNilMsg error = grpcstatus.Errorf(grpccodes.InvalidArgument, "nil msg")
+
+func (k Keeper) CreateDenom(
```

Comment:
I think we created this without thinking the use case and how it aligns with our chain.